### PR TITLE
Person Alias, Vessel Permissions, and related changes

### PR DESCRIPTION
**Pull request by sethgerou-noaa**
Creates PersonAlias and replaces Person with PersonAlias in all relevant places. 
Note: PersonAlias contains a roles array - so active Ashop Observers can be found by: 
isActive, isAshop, and roles.includes('observer').  
 
Removes Person (Captain) from Vessel - captain person is now tied to a vessel through vessel-permissions. The Person doc can continue to store full (Active) Vessel, as Captain Person is no longer stored on Vessel.

Adds passCode to vessel - this is a code an authorized user can supply to gain access to a vessel.

Adds VesselPermissons doc - stores the ids of authorized users per vessel.  will be a lookup.  storing ids only to keep this file as small as possible.

Making these changes impacted several docs.  Please review carefully and let me know of any concerns/anything that calls for discussion.

Thanks!

### CODE REVIEW

File: bn-models/src/models/_lookups/vessel.ts [hunk 1]
```diff
@@ -35,12 +34,12 @@ export interface Vessel extends BaseLookup {
   stateRegulationNumber?: string;
   registeredLength?: Measurement;
   vesselStatus?: string; // lookup value - Active, Inactive, Retired, Sank, Byback
-  captains?: VesselCaptain[];
   safetyMeasures?: SafetyMeasure[]; // 30+ items
   media?: Media[];
   isActive?: boolean;
   emHardware?: string;
   thirdPartyReviewer?: string;
+  passCode?: string // user must supply correct vessel passcode to gain access to vessel.
```
Comment (wsmith-nwfsc) [inline bn-models/src/models/_lookups/vessel.ts]: Interesting.

Comment (SaOgaz-NOAA) [reply bn-models/src/models/_lookups/vessel.ts]: Yeah, this is what OLE does currently in their over the phone system. Not sure why we didn't think to apply it to our system sooner... I think it makes a lot of sense.

**Comment by SaOgaz-NOAA** [comment]
This looks great to me @sethgerou-noaa! I can't currently think of any problems from my end with the OLE stuff. Although I guess this way we still have to scrape the vesselpermissions doc for users as the top level of the doc is organized by vessel. Does having the top org level be vessels (as oppose to the top level being people and then authorized vessels under that) have some benefits for the way you need to use this information? I think for my end of the app I'll only ever be using this doc indirectly for the part where we will populate the active vessel drop down choices.

Separate comment, do you foresee us being able to have just "one" authorization/approval for making trips and making declarations? I know we've talked about this but just wanted to be 100% sure.  It seems like the observer side is the more strict one on this, so hoping they're also good with the passcode idea.

**Comment by sethgerou-noaa** [comment]
> This looks great to me @sethgerou-noaa! I can't currently think of any problems from my end with the OLE stuff. Although I guess this way we still have to scrape the vesselpermissions doc for users as the top level of the doc is organized by vessel. Does having the top org level be vessels (as oppose to the top level being people and then authorized vessels under that) have some benefits for the way you need to use this information? I think for my end of the app I'll only ever be using this doc indirectly for the part where we will populate the active vessel drop down choices.
> 
> Separate comment, do you foresee us being able to have just "one" authorization/approval for making trips and making declarations? I know we've talked about this but just wanted to be 100% sure. It seems like the observer side is the more strict one on this, so hoping they're also good with the passcode idea.

I guess to me it makes somewhat more sense to associate a collection of users to a vessel than a collection of vessels to a user.  Seems like we'll have to do a little data manipulation in either approach, as some screens like trips and declarations expose vessels a user has access to, and other screens like vessel management/details expose users that have access to a vessel.  

As permissions are contained in a single doc, it should be pretty simple to flatten vessel/person-alias associations, then filter for the active user:  (something like...)

const permissions = {};

for (vessel of vesselAuthorizations) {
  for (person of authorizedPeople) {
    permissions.push({vesselId: vessel.vesselId, vesselIndex: vesselAuthorizations.indexOf(vessel), person-alias-id: person })
  }
}

const userVesselPermissions = permissions.filter( (permission) => permission.person-alias-id === [activeuser persion-alias-id] )

Yeah, I'm really hoping we can have just one approval process - and will certainly attempt to get buy in from the observer staff on the passcode concept.

**Pull request merged**